recurly/pager.py

```diff
@@ -82,9 +87,11 @@ def take(self, n):
         """Performs a request with the pager `limit` set to `n` and only returns the
         first `n` results in the response. `n` is limited to the maximum page size.
         """
-        params = self.__params.copy()
-        params.update({"limit": n})
-        items = PageIterator(self.__client, self.__path, params)
+        options = self.__options
+        if "params" in options:
+            options = copy.deepcopy(options)
```

Comment:
Might be able to get away with not doing the deepcopy, but thought we might want to be safe here.